src/Host/Broker/Impl/Sessions/SessionsController.cs

```diff
@@ -40,12 +44,18 @@ public class SessionsController : Controller {
 
             string profilePath = User.FindFirst(Claims.RUserProfileDir)?.Value;
 
-            var interp = string.IsNullOrEmpty(request.InterpreterId)
-                ? _interpManager.Interpreters.First()
-                : _interpManager.Interpreters.First(ip => ip.Id == request.InterpreterId);
+            Interpreter interp;
+            if (!string.IsNullOrEmpty(request.InterpreterId)) {
+                interp = _interpManager.Interpreters.FirstOrDefault(ip => ip.Id == request.InterpreterId);
+                if (interp == null) {
+                    return Task.FromResult(CustomErrorResult.Create(CustomHttpError.InterpreterNotFound));
+                }
+            } else {
+                interp = _interpManager.Interpreters.First();
+            }
 
             var session = _sessionManager.CreateSession(User.Identity, id, interp, securePassword, profilePath, request.CommandLineArguments);
-            return Task.FromResult(session.Info);
+            return Task.FromResult(new ObjectResult(session.Info) as IActionResult);
```

Comment:
Nit: `FromResult<IActionResult>` (to avoid a runtime cast, and enforce static verification).